pkg/workload/workload.go

```diff
@@ -69,19 +71,39 @@ func (psr *PodSetResources) ScaledTo(newCount int32) *PodSetResources {
 	return ret
 }
 
-func NewInfo(w *kueue.Workload) *Info {
+func NewInfo(cl client.Client, w *kueue.Workload) *Info {
 	info := &Info{
 		Obj: w,
 	}
+	info.TotalPodTemplates = podTemplateSpecs(cl, w)
 	if w.Status.Admission != nil {
 		info.ClusterQueue = string(w.Status.Admission.ClusterQueue)
 		info.TotalRequests = totalRequestsFromAdmission(w)
 	} else {
-		info.TotalRequests = totalRequestsFromPodSets(w)
+		info.TotalRequests = totalRequestsFromPodSets(w, info.TotalPodTemplates)
 	}
 	return info
 }
 
+func podTemplateSpecs(cl client.Client, wl *kueue.Workload) map[string]corev1.PodTemplateSpec {
+	podTempateRefs := make(map[string]string, 0)
+	for _, p := range wl.Spec.PodSets {
+		if p.PodTemplateRef == nil {
+			continue
+		}
+		podTempateRefs[p.PodTemplateRef.Name] = p.PodTemplateRef.Namespace
+	}
+	podTemplateSpecs := make(map[string]corev1.PodTemplateSpec, 0)
+	for name, namespace := range podTempateRefs {
+		var podTemplate corev1.PodTemplate
+		if err := cl.Get(context.Background(), client.ObjectKey{Name: name, Namespace: namespace}, &podTemplate); err != nil {
```

Comment:
What happens in the Template object doesn't exist? is there a mechanism to retry when the template object appears?